common/lib/plugins/failover/reader_failover_handler.ts

```diff
@@ -322,6 +322,21 @@ class ConnectionAttemptTask {
     );
     try {
       this.targetClient = await this.pluginService.forceConnect(this.newHost, copy);
+
+      // ensure that new connection is a connection to a reader node
+      try {
+        // eslint-disable-next-line no-constant-condition
+        while (1) {
```

Comment:
It might be safer to have an end condition. A timeout task will be raced against a different task involving this call, but at the moment if the loop keeps going it doesn't seem like there's a way to stop and finish the task.